### PR TITLE
feat: `avm/res/cache/redis`

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.cache.redis](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.cache.redis.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Requires uncommenting in the `/.github/CODEOWNERS` file

**Comment by segraef** [comment]
> Requires uncommenting in the `/.github/CODEOWNERS` file

done

File: avm/res/cache/redis/main.bicep [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+  'Standard'
+])
+@description('Optional. The type of Redis cache to deploy.')
+param skuName string = 'Basic'
+
+@description('Optional. Static IP address. Optionally, may be specified when deploying a Redis cache inside an existing Azure Virtual Network; auto assigned by default.')
+param staticIP string = ''
+
+@description('Optional. The full resource ID of a subnet in a virtual network to deploy the Redis cache in. Example format: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1.')
+param subnetId string = ''
```
Comment (AlexanderSehr) [inline avm/res/cache/redis/main.bicep:88]:
```suggestion
@description('Optional. The full resource ID of a subnet in a virtual network to deploy the Redis cache in.')
param subnetResourceId string = ''
```

Note: Needs updating in other locations & regenerations of files

File: avm/res/cache/redis/main.bicep [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+@description('Optional. If the zoneRedundant parameter is true, replicas will be provisioned in the availability zones specified here. Otherwise, the service will choose where replicas are deployed.')
+param zones array = []
+
+@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints privateEndpointType
+
+@description('Optional. The diagnostic settings of the service.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
```
Comment (AlexanderSehr) [inline avm/res/cache/redis/main.bicep:105]:
```suggestion
@description('Optional. Enable/Disable usage telemetry for module.')
```

File: avm/res/cache/redis/main.bicep [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+output hostName string = redis.properties.hostName
+
+@description('Redis SSL port.')
+output sslPort int = redis.properties.sslPort
+
+@description('The full resource ID of a subnet in a virtual network where the Redis Cache was deployed in.')
+output subnetId string = !empty(subnetId) ? redis.properties.subnetId : ''
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) && contains(redis.identity, 'principalId') ? redis.identity.principalId : ''
```
Comment (AlexanderSehr) [inline avm/res/cache/redis/main.bicep:269]:
```suggestion
output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) ? redis.identity.?principalId : ''
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -18,7 +18,7 @@
 #/avm/res/authorization/role-definition/ @Azure/avm-res-authorization-roledefinition-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/automation/automation-account/ @Azure/avm-res-automation-automationaccount-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/batch/batch-account/ @Azure/avm-res-batch-batchaccount-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/cache/redis/ @Azure/avm-res-cache-redis-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/cache/redis/ @Azure/avm-res-cache-redis-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (AlexanderSehr) [inline .github/CODEOWNERS:21]: Team must be created

File: .github/workflows/avm.res.cache.redis.yml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  ###########################
+  #   Initialize pipeline   #
+  ###########################
+  job_initialize_pipeline:
+    runs-on: ubuntu-20.04
```
Comment (eriqua) [inline .github/workflows/avm.res.cache.redis.yml:47]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/CODEOWNERS (hunk 4 above)
Comment (segraef) [reply .github/CODEOWNERS:21]: @Azure/avm-res-cache-redis-module-owners-bicep
@Azure/avm-res-cache-redis-module-contributors-bicep

File: avm/res/cache/redis/main.bicep [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+output hostName string = redis.properties.hostName
+
+@description('Redis SSL port.')
+output sslPort int = redis.properties.sslPort
+
+@description('The full resource ID of a subnet in a virtual network where the Redis Cache was deployed in.')
+output subnetResourceId string = !empty(subnetResourceId) ? redis.properties.subnetResourceId : ''
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) ? redis.identity.?principalId : ''
```
Comment (AlexanderSehr) [inline avm/res/cache/redis/main.bicep:269]:
```suggestion
output systemAssignedMIPrincipalId string = redis.?identity.?principalId ?? ''
```

File: .github/CODEOWNERS (hunk 4 above)
Comment (AlexanderSehr) [reply .github/CODEOWNERS:21]: It still shows errors, as the link to the parent team may not be established/approved yet

File: avm/res/cache/redis/tests/e2e/max/main.test.bicep [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    location: location
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
```
Comment (AlexanderSehr) [inline avm/res/cache/redis/tests/e2e/max/main.test.bicep:63]: Missing RBAC test. Please add, test & update the generated files

Comment (AlexanderSehr) [reply avm/res/cache/redis/tests/e2e/max/main.test.bicep:63]: Could even be added to the Private Endpointpoint deployment

File: .github/CODEOWNERS (hunk 4 above)
Comment (segraef) [reply .github/CODEOWNERS:21]: both teams exist and parents are assigned

File: avm/res/cache/redis/tests/e2e/max/main.test.bicep (hunk 7 above)
Comment (segraef) [reply avm/res/cache/redis/tests/e2e/max/main.test.bicep:63]: Let's just leave the RBACA test for now ;), done.

Comment (segraef) [reply avm/res/cache/redis/tests/e2e/max/main.test.bicep:63]: done, just triggered a fresh workflow run

File: avm/res/cache/redis/main.bicep [hunk 8]
```diff
@@ -0,0 +1,422 @@
...
+          privateLinkServiceId: redis.id
+          groupIds: [
+            privateEndpoint.?service ?? 'redisCache'
+          ]
+        }
+      }
+    ]
+    name: privateEndpoint.?name ?? 'pep-${last(split(redis.id, '/'))}-${privateEndpoint.?service ?? 'redisCache'}-${index}'
+    subnetResourceId: privateEndpoint.subnetResourceId
+    enableTelemetry: enableTelemetry
```
Comment (AlexanderSehr) [inline avm/res/cache/redis/main.bicep:235]:
```suggestion
    enableTelemetry: privateEndpoint.?enableTelemetry ?? enableTelemetry
```

File: avm/res/cache/redis/main.json [hunk 9]
```diff
@@ -0,0 +1,1454 @@
...
+            ]
+          },
+          "name": {
+            "value": "[coalesce(tryGet(coalesce(parameters('privateEndpoints'), createArray())[copyIndex()], 'name'), format('pep-{0}-{1}-{2}', last(split(resourceId('Microsoft.Cache/redis', parameters('name')), '/')), coalesce(tryGet(coalesce(parameters('privateEndpoints'), createArray())[copyIndex()], 'service'), 'redisCache'), copyIndex()))]"
+          },
+          "subnetResourceId": {
+            "value": "[coalesce(parameters('privateEndpoints'), createArray())[copyIndex()].subnetResourceId]"
+          },
+          "enableTelemetry": {
+            "value": "[parameters('enableTelemetry')]"
```
Comment (AlexanderSehr) [inline avm/res/cache/redis/main.json:761]: Apperas outdated. Please refresh the `main.json` given the latest change

Comment (segraef) [reply avm/res/cache/redis/main.json]: done

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**